activerecord/test/models/comment.rb

```diff
@@ -45,6 +45,10 @@ class SubSpecialComment < SpecialComment
 class VerySpecialComment < Comment
 end
 
+class IdDescOrderedComment < Comment
```

Comment:
For new tests we try not to expand our global models but describe the specific situation right inside the test-case. This could look something like:

```ruby
 def test_preloading_has_one_using_reorder
    klass = Class.new(ActiveRecord::Base) do
      def self.name; "TempAuthor"; end
      self.table_name = "authors"
      has_one :post, class_name: "PostWithDefaultScope", foreign_key: :author_id
      has_one :reorderd_post, -> { reorder(title: :desc) }, class_name: "PostWithDefaultScope", foreign_key: :author_id
    end

    author = klass.first
    # PRECONDITION: make sure ordering results in different results
    assert_not_equal author.post, author.reorderd_post

    preloaded_reorderd_post = klass.preload(:reorderd_post).first.reorderd_post

    assert_equal author.reorderd_post, preloaded_reorderd_post
    assert_equal posts(:sti_post_and_comments).title, preloaded_reorderd_post.title
  end
```